src/js/map/MapLoading.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+
+// Required event dispatch and subscription for now while
+// parts of app are React components and others are not
+import { EventEmitter } from '../components/event-emitter';
+
+export default class MapLoading extends React.Component {
+    constructor (props) {
+        super(props);
+
+        this.state = {
+            loading: false
+        };
+    }
+
+    componentDidMount () {
+        // When a Modal component fires "on", shield turns itself on.
+        EventEmitter.subscribe('maploading:on', () => {
+            this.setState({ visible: true });
+        });
+
+        // When a Modal component fires "off", shield turns itself off.
+        EventEmitter.subscribe('maploading:off', () => {
```

Comment:
That's a good idea. In any case, I'm hoping this particular one goes away at some point when we start thinking about this component responding to app state rather than events.